### PR TITLE
Blog-App Add devise

### PR DESCRIPTION
**Pull request by Piercel2022**
In this project, 

- I add the devise gem to the app and use it for the registration and login of users.

- I Remove current_user method in ApplicationController, devise will provide us with one.
- I install and set up the devise's gem.
- Can register a new user.
- User logs in with a combination of email and password.

- Hashed password should be stored in the database.

- Ask for confirmation of email.

- Can reset password.

### CODE REVIEW

File: app/views/devise/sessions/new.html.erb [hunk 1]
```diff
@@ -0,0 +1,26 @@
...
+      <%= f.label :remember_me %>
+    </div>
+  <% end %>
+
+  <div class="actions">
+    <%= f.submit "Log in" %>
+  </div>
+<% end %>
+
+<%= render "devise/shared/links" %>
```
Comment (CollinsTatang) [inline app/views/devise/sessions/new.html.erb]: Hi @Piercel2022 and @MudasirSherwani 

- [ ] Please your devise view for `Registration` and `Login` should be modified to match the given [wireframe](https://github.com/microverseinc/curriculum-rails/blob/main/blog-app/sneak_peek.md)
image

File: app/models/user.rb [hunk 2]
```diff
@@ -1,4 +1,10 @@
 class User < ApplicationRecord
+  # Include default devise modules. Others available are:
+  # :confirmable, :lockable, :timeoutable, :trackable and :omniauthable
+  devise :database_authenticatable, :registerable,
+         :recoverable, :rememberable, :validatable
```
Comment (CollinsTatang) [inline app/models/user.rb]: **Duplication**
- [ ] Please remove this part of the code we already have it properly enable below :point_down:

**Review by CollinsTatang — CHANGES_REQUESTED** [review]
Hi @Piercel2022 and @MudasirSherwani, 

Good job so far!
There are some issues that you still need to work on to go to the next project but you are almost there!

### **Highlights 🚀 :heavy_check_mark: :heavy_check_mark:** 

- The repo looks professional.:heavy_check_mark:
- Git/GitHub flow best practices have been followed:heavy_check_mark:
- There are no linter errors:heavy_check_mark:
- You have descriptive PR:heavy_check_mark:

## Required changes 
It's ok to have bugs in your code. Please, see this code review as a chance for you to learn practical coding skills and learn new tips that can improve your code quality and help you get a better job.
_Check the comments under the review._
## Optional suggestions
_Every comment with the [OPTIONAL] prefix is not crucial enough to stop the approval of this PR. However, I strongly recommend you to take them into account as they can make your code better._

Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please, remember to tag @CollinsTatang in your question so I can receive the notification.

**_Please, do not open a new Pull Request for re-reviews. You should use the same Pull Request submitted for the first review, either valid or invalid unless it is requested otherwise._**

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561) you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

File: app/views/devise/registrations/new.html.erb [hunk 3]
```diff
@@ -0,0 +1,29 @@
...
+
+  <div class="field">
+    <%= f.label :password_confirmation %><br />
+    <%= f.password_field :password_confirmation, autocomplete: "new-password" %>
+  </div>
+
+  <div class="actions">
+    <%= f.submit "Sign up" %>
+  </div>
+<% end %>
```
Comment (tugsem) [inline app/views/devise/registrations/new.html.erb]: - [ ] Kindly fix the layout of the form elements to match the wireframe as the prev reviewer asked, make sure ensure it's responsive. It looks like this:
image

File: app/views/devise/sessions/new.html.erb [hunk 4]
```diff
@@ -0,0 +1,26 @@
...
+
+  <% if devise_mapping.rememberable? %>
+    <div class="field">
+      <%= f.check_box :remember_me %>
+      <%= f.label :remember_me %>
+    </div>
+  <% end %>
+
+  <div class="actions">
+    <%= f.submit "Log in" %>
```
Comment (tugsem) [inline app/views/devise/sessions/new.html.erb]: - [ ] Kindly fix the layout of the form elements to match the wireframe as the prev reviewer asked, make sure ensure it's responsive. It looks like this:
image
- I also noticed the edit user form page has the same problem. Kindly fix that too.

**Review by tugsem — CHANGES_REQUESTED** [review]
## STATUS: REQUIRED CHANGES ♻️
Hello @Piercel2022  and @CollinsTatang 👋 , 

Great work on making the changes requested by a previous reviewer 👏 
You've done well implementing some of the requested changes, but there are some issues that you still need to work on to go to the next project but you are almost there!

Alt Text

 ## Highlights 🚀
✔️Good job on applying functionality of devise gem.

 ## Required Changes ♻️
_Check the comments under the review._

## Optional suggestions

_Every comment with the [OPTIONAL] prefix won't stop the approval of this PR. However, I strongly recommend you to take them into account as they can make your code better. Some of them were simply missed by the previous reviewer and addressing them will really improve your application._


Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please, remember to tag me in your question so I can receive the notification.

**_Please, do not open a new Pull Request for re-reviews. You should use the same Pull Request submitted for the first review, either valid or invalid unless it is requested otherwise._**


------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561)  you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

**Review by zainsadaqat — APPROVED** [review]
# Pull Request Approved ✅ 🎊 🎉 
## Hello @Piercel2022 👋 

## Things to Appreciate 
- [ ] Correct Flow ✅ 
- [ ] Clean Code 👍 
- [ ] No Linter Errors 🔥 
- [ ] Descriptive Pull Request and Summary 👌 

### Your project is complete!  There is nothing else to say other than it's time to merge it.

# Congratulations!!! 🎉 🥳 👏 🙌 🎊 ☑️ 



Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please, remember to tag me in your question so I can receive the notification.

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561)  you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

**Pull request merged**